api/turing/cluster/controller.go

```diff
@@ -301,6 +302,26 @@ func (c *controller) DeleteKnativeService(
 	return services.Delete(ctx, svcName, metav1.DeleteOptions{})
 }
 
+// GetKnativeServiceDesiredReplicas determines the current desired number of replicas for the default revision
+func (c *controller) GetKnativeServiceDesiredReplicas(
+	ctx context.Context,
+	svcName string,
+	namespace string,
+) (int, error) {
+	// Init knative revisions getter
+	revisions := c.knServingClient.Revisions(namespace)
+
+	rev, err := revisions.Get(ctx, getDefaultRevisionName(svcName), metav1.GetOptions{})
+	if err != nil {
+		return 0, err
+	}
+	if rev.Status.DesiredReplicas != nil {
+		return int(*rev.Status.DesiredReplicas), nil
+	}
+
+	return 0, fmt.Errorf("Desired Replicas for %s/%s is not set", namespace, svcName)
```

Comment:
nit: I think it's better to switch it so that we handle and return invalid case first:

```
	if rev.Status.DesiredReplicas == nil {
		return 0, fmt.Errorf("Desired Replicas for %s/%s is not set", namespace, svcName)
	}

	return int(*rev.Status.DesiredReplicas), nil
```